src/core/services/interimElement/interimElement.js

```diff
@@ -323,30 +329,40 @@ function InterimElementProvider() {
        */
       function hide(reason, options) {
         if ( !stack.length ) return $q.when(reason);
+        if (hideQueue >= stack.length) return;
         options = options || {};
 
         if (options.closeAll) {
-          var promise = $q.all(stack.reverse().map(closeElement));
-          stack = [];
-          return promise;
+          return $q.all(stack.reverse().map(closeElement));
         } else if (options.closeTo !== undefined) {
```

Comment:
Updated, but I don't get what you mean with the else statement?

```
if (options.closeAll) {
  return $q.all(stack.reverse().map(closeElement));
} else if (options.closeTo) {
  return $q.all(stack.slice(options.closeTo).map(closeElement));
} else {
  return closeElement(stack[hideQueue]);
}
```
Every check will be used. Mostly this `return closeElement(stack[hideQueue]);` when showing / hiding a toast